src/global_test/mod.rs

```diff
@@ -0,0 +1,239 @@
+//! # Test module
+//!
+//! This module contain all the initialization and global functions needed by the tests
+#[cfg(test)]
+pub mod tests {
+    use crate::{model, regiment};
+
+    /// # Initialize the stats structs with some values given as parameters
+    ///
+    /// # Parameters
+    /// advance (usize): the advance statistic of the model
+    ///
+    /// march (usize): the march statistic of the model
+    ///
+    /// discipline (usize): the discipline statistic of the model
+    ///
+    /// health_point (usize): the health_point statistic of the model
+    ///
+    /// defense (usize): the defense statistic of the model
+    ///
+    /// resilience (usize): the resilience statistic of the model
+    ///
+    /// armour (usize): the armour statistic of the model
+    ///
+    /// aegis (usize): the aegis statistic of the model
+    ///
+    /// attack (usize): the attack statistic of the model
+    ///
+    /// offensive (usize): the offensive statistic of the model
+    ///
+    /// strength (usize): the strength statistic of the model
+    ///
+    /// armour_penetration (usize): the armour_penetration statistic of the model
+    ///
+    /// agility (usize): the agility statistic of the model
+    /// # Return
+    /// Stats Model: The Stats Model created and initialized
+
+    pub fn initialize_stats(
+        advance: usize,
+        march: usize,
+        discipline: usize,
+        health_point: usize,
+        defense: usize,
+        resilience: usize,
+        armour: usize,
+        aegis: usize,
+        attack: usize,
+        offensive: usize,
+        strength: usize,
+        armour_penetration: usize,
+        agility: usize,
+    ) -> model::Stats {
+        let stats: model::Stats = model::Stats::new(
+            model::GlobalStats {
+                advance: advance,
+                march: march,
+                discipline: discipline,
+            },
+            model::DefensiveStats {
+                health_point: health_point,
+                defense: defense,
+                resilience: resilience,
+                armour: armour,
+                aegis: aegis,
+            },
+            model::OffensiveStats {
+                attack: attack,
+                offensive: offensive,
+                strength: strength,
+                armour_penetration: armour_penetration,
+                agility: agility,
+            },
+        );
+        stats
+    }
+
+    /// # Initialize all stats structs values to 1
+    ///
+    /// # Return
+    /// Stats Model: The Stats Model created and initialized
```

Comment:
"Our mock Stats struct" seems a better description for the "Return" field
because "The Stats Model"  kindof disturb me